### DISCUSSION TITLE
ScriptReference/SerializedProperty-editable

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#596] [Notes]
# ScriptReference/SerializedProperty-editable

https://docs.unity3d.com/ScriptReference/SerializedProperty-editable.html

**Comment by vertxxyz** [comment]
If you find a SerializedProperty is not editable, you may have created a [`SerializedObject`](https://docs.unity3d.com/ScriptReference/SerializedObject.html) out of an [`Editor`](https://docs.unity3d.com/ScriptReference/Editor.html), and it's [`hideFlags`](https://docs.unity3d.com/ScriptReference/Object-hideFlags.html) are marked as [`HideAndDontSave`](https://docs.unity3d.com/ScriptReference/HideFlags.HideAndDontSave.html).
To fix this and make the property editable, you can remove the Editor's [`NotEditable`](https://docs.unity3d.com/ScriptReference/HideFlags.NotEditable.html) flag before creating the `SerializedObject`.
```csharp
hideFlags &= ~HideFlags.NotEditable;
```
